### DISCUSSION TITLE
How to get the distance information between Loco Positioning Nodes?

### DISCUSSION THREAD

**Discussion by Guitarwang01** [bitcraze/discussions#991] [Q&A] [answered]
Hi! I want to ask something about the usage of Loco Positioning Nodes.
We want to install the Loco Positioning Nodes in cars for mapping work, but we didn't find relative APIs to get the distance between nodes. Is there any methods to do it?

**Comment by krichardsson** [comment]
Could you tell us a bit more about your setup, are you only using nodes or do you also use a Crazyflie somewhere? 
Where do you want to get the distance information? In the nodes, in the Crazyflie (if you have one), in a ground station?
How do you want to use the distance information?

**Reply by Guitarwang01** [reply]
We proposed to use several cars carrying LiDAR or something for mapping automatically and construct complete map information. So we plan to equip them with the Loco Positioning Nodes to do UWB positioning for each other. Crazyflie will also carry the node and cameras to make a full map with UWB positioning and image stitching. The distance information is used in UWB positioning for the cars.

**Reply by krichardsson** [reply, marked as answer]
Thanks for the information!
I will assume you have some sort of device in the car that can connect to the node using USB. There are a lot of similarities with #775, you might get some ideas there.

My suggestion is that you run the system in TDoA3 mode as the nodes are doing Two Way Ranging with each other in this mode. The only small hurdle is to get the distance data out from the anchors, but it should require very little work to add functionality to stream it out on the USB/serial port.

**Reply by Guitarwang01** [reply]
Thank you for your reply! We will try this by study other ideas!
By the way, I also want to ask about whether it is OK to get distance information of two drones carrying the Loco Pos Deck?

**Reply by knmcguire** [reply]
Perhaps take a look at this old discussion: https://github.com/orgs/bitcraze/discussions/147. @krichardsson were there any updates here as of recently?

**Comment by krichardsson** [comment]
> whether it is OK to get distance information of two drones carrying the Loco Pos Deck

In theory yes, but there is no code for it in the Crazyflie. In TDoA3 mode the CF is only receiving and to get the distance to other CFs they must also transmit. This is an old idea that we never implemented (we call it hybrid mode) and I started an [example implementation](https://gist.github.com/krichardsson/c1a51642d2d560d825f7bee8577ac6c6) but as far as I remember it is not fully functional and should be seen as a suggestion of where to add the transmit functionality rather than a working implementation.

There is also some discussions on this topic in https://github.com/bitcraze/crazyflie-firmware/issues/576

**Reply by Guitarwang01** [reply]
Thank you very much!
